tests/v2/validation/scaling/custom_cluster_nodes.go

```diff
@@ -0,0 +1,86 @@
+package scaling
+
+import (
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	clusterNameLabel = "cluster.x-k8s.io/cluster-name"
+	workloadNS       = "default"
+	clusterNS        = "fleet-default"
+)
+
+func validateNodeCount(client *rancher.Client, clusterID string, nodeType string, initialNodeCount int) (bool, error) {
```

Comment:
I would think it makes more sense here to have parameters similar to `MatchNodeToRole`, where the roles are booleans instead of 1 string. This would greatly reduce the copy-paste of this function, as you could directly pass them to the node.List call and remove all the if/else statements.